task2_api/task2_api.py

```diff
@@ -0,0 +1,115 @@
+import requests
+from collections import Counter
+import copy
+from datetime import datetime, timedelta
+from decimal import Decimal
+import csv
+
+
+class MovieData:
+
+    def __init__(self, num_pages):
+        self.headers = {
+            "accept": "application/json",
+            "Authorization": "Bearer eyJhbGciOiJIUzI1NiJ9.eyJhdWQiOiIzMTI3NGFmYTRlNTUyMjRjYzRlN2Q0NmNlMTNkOTZjOSIsInN1YiI6IjVkNmZhMWZmNzdjMDFmMDAxMDU5NzQ4OSIsInNjb3BlcyI6WyJhcGlfcmVhZCJdLCJ2ZXJzaW9uIjoxfQ.lbpgyXlOXwrbY0mUmP-zQpNAMCw_h-oaudAJB6Cn5c8"
+        }
+        self.num_pages = str(num_pages)
+        self.movie_data = None
+        self.sorted_movie_data = None
+        # self.genres
+
+    def get_movie_data(self):
+        url1 = f"https://api.themoviedb.org/3/discover/movie?include_adult=false&include_video=false&sort_by=popularity.desc&page={self.num_pages}"
+        response = requests.get(url1, headers=self.headers)
+        if response.status_code == 200:
+            self.movie_data = response.json()['results']
+        else:
+            print("Error fetching movie data.")
+
+    def get_unique_genres(self):
+        url2 = f"https://api.themoviedb.org/3/genre/movie/list?language=en"
+        response_g = requests.get(url2, headers=self.headers)
+        if response_g.status_code == 200:
+            genre_data = response_g.json()
+            genres = {}
+            for genre in genre_data['genres']:
+                genre_id = genre['id']
+                genre_name = genre['name']
+                genres[genre_name] = genre_id
+            return genres
+        else:
+            print("Error fetching genre data.")
```

Comment:
as you have return in the **if** block you need to add **return** here or use **rise**